engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -302,19 +309,77 @@ private class Factory(
         val location = makeFieldCoordinates(parentType, hydratedFieldDef)
 
         val batchSize = hydration.batchSize ?: 50
+        val hydrationArgs = getHydrationArguments(hydration, parentType, hydratedFieldDef, actorFieldDef)
+
+        val matchStrategy = if (hydration.isObjectMatchByIndex) {
+            NadelBatchHydrationMatchStrategy.MatchIndex
+        } else if (hydration.objectIdentifiers.isNotEmpty()) {
```

Comment:
So reminder `hydration` here is a `UnderlyingServiceHydration` which is the output type from the parser.

`objectIdentifiers` here is the new `inputIdentifiedBy` syntax. I've deprecated the old `objectIdentifier` field.